src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph_RemoveProject.cs

```diff
@@ -124,25 +129,30 @@ private static ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> Compu
     /// <seealso cref="ComputeNewReverseTransitiveReferencesMapForRemovedProject"/>
     private static ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> ComputeNewTransitiveReferencesMapForRemovedProject(
         ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> existingTransitiveReferencesMap,
-        ProjectId removedProjectId)
+        ArrayBuilder<ProjectId> removedProjectIds)
     {
         var builder = existingTransitiveReferencesMap.ToBuilder();
 
         // Iterate over each project and invalidate the transitive references for the project if the project has an
         // existing transitive reference to 'removedProjectId'.
         foreach (var (project, references) in existingTransitiveReferencesMap)
         {
-            if (references.Contains(removedProjectId))
+            foreach (var removedProjectId in removedProjectIds)
             {
-                // The project transitively referenced 'removedProjectId', so any transitive references brought in
-                // exclusively through this reference are no longer valid. Remove the project from the map and the
-                // new transitive references will be recomputed the first time they are needed.
-                builder.Remove(project);
+                if (references.Contains(removedProjectId))
+                {
+                    // The project transitively referenced 'removedProjectId', so any transitive references brought in
+                    // exclusively through this reference are no longer valid. Remove the project from the map and the
+                    // new transitive references will be recomputed the first time they are needed.
+                    builder.Remove(project);
+                }
```

Comment:
break at this point, since once we're called builder.Remove() there's no reason to be checking any further.